lambdas/check-gpg45-score/src/main/java/uk/gov/di/ipv/core/checkgpg45score/CheckGpg45ScoreHandler.java

```diff
@@ -0,0 +1,143 @@
+import static uk.gov.di.ipv.core.library.journeyuris.JourneyUris.JOURNEY_ERROR_PATH;
+import static uk.gov.di.ipv.core.library.journeyuris.JourneyUris.JOURNEY_MET_PATH;
+import static uk.gov.di.ipv.core.library.journeyuris.JourneyUris.JOURNEY_UNMET_PATH;
+
+public class CheckGpg45ScoreHandler implements RequestHandler<ProcessRequest, Map<String, Object>> {
+    private final ConfigService configService;
+    private final ClientOAuthSessionDetailsService clientOAuthSessionDetailsService;
+    private final Gpg45ProfileEvaluator gpg45ProfileEvaluator;
+    private final IpvSessionService ipvSessionService;
+    private final UserIdentityService userIdentityService;
+    private final String componentId;
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    @SuppressWarnings("unused") // Used by tests through injection
+    public CheckGpg45ScoreHandler(
+            ConfigService configService,
+            ClientOAuthSessionDetailsService clientOAuthSessionDetailsService,
+            Gpg45ProfileEvaluator gpg45ProfileEvaluator,
+            IpvSessionService ipvSessionService,
+            UserIdentityService userIdentityService) {
+        this.configService = configService;
+        this.clientOAuthSessionDetailsService = clientOAuthSessionDetailsService;
+        this.gpg45ProfileEvaluator = gpg45ProfileEvaluator;
+        this.ipvSessionService = ipvSessionService;
+        this.userIdentityService = userIdentityService;
+
+        componentId = configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID);
+    }
+
+    @SuppressWarnings("unused") // Used by AWS
+    @ExcludeFromGeneratedCoverageReport
+    public CheckGpg45ScoreHandler() {
+        this.configService = new ConfigService();
+        this.clientOAuthSessionDetailsService = new ClientOAuthSessionDetailsService(configService);
+        this.gpg45ProfileEvaluator = new Gpg45ProfileEvaluator(configService);
+        this.ipvSessionService = new IpvSessionService(configService);
+        this.userIdentityService = new UserIdentityService(configService);
+
+        componentId = configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public Map<String, Object> handleRequest(ProcessRequest event, Context context) {
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(event);
+            String featureSet = RequestHelper.getFeatureSet(event);
+            String scoreType = RequestHelper.getScoreType(event);
+            Integer scoreThreshold = RequestHelper.getScoreThreshold(event);
+            configService.setFeatureSet(featureSet);
+
+            List<SignedJWT> credentials = getParsedCredentials(ipvSessionId);
+            Gpg45Scores gpg45Scores = gpg45ProfileEvaluator.buildScore(credentials);
+            int fraudScore = gpg45Scores.getFraud();
+
+            if (fraudScore >= scoreThreshold) {
```

Comment:
Think this might need making abstract - we're passing in the scoreType that we're interested in, but hard coding to fraud.